Update roadmap to -terms instead of quarters

EmLauber opened the pull request:
Changing roadmap to be a near-, mid-, and long- term style instead of quarterly.

EmLauber committed:
Update roadmap to -terms instead of quarters

Changing roadmap to be a near-, mid-, and long- term style instead of quarterly.

hectormmg commented on roadmap.md:
```diff
@@ -1,60 +1,41 @@
+- Ivy Library Distribution of MSAL-angular 
+- Angular 14 support 
+- Sustainable strategy to release adoption drivers like library dependency changes  
+- Software bound RT Pop support  
```
```suggestion
- Software Bound Refresh Token Proof-of-Possession Support
```

tnorling commented on roadmap.md:
```diff
@@ -1,60 +1,41 @@
 # Roadmap
 
-The MSAL.js team schedules work on a quarterly basis. The roadmap below provides a subset of the team's quarterly commitments for past and upcoming quarters. 
+The MSAL.js team's roadmap is presented as near-term, mid-term, & long-term to give a people a better idea of what we are currently working on, what is up next for us, and what we are considering for the future. These are not guarenteed timelines, but insight into the team's current direction. The roadmap is focused on big features or focus areas; it may not call out each individual work item tha ...
```
```suggestion
The MSAL.js team's roadmap is presented as near-term, mid-term, & long-term to give you a better idea of what we are currently working on, what is up next for us, and what we are considering for the future. These are not guaranteed timelines, but instead intended to provide insight into the team's current direction. The roadmap is focused on big features or focus areas; it may not call out each individual work item that is added to the library. 
```

tnorling commented on roadmap.md:
```diff
@@ -1,60 +1,41 @@
+The MSAL.js team's roadmap is presented as near-term, mid-term, & long-term to give a people a better idea of what we are currently working on, what is up next for us, and what we are considering for the future. These are not guarenteed timelines, but insight into the team's current direction. The roadmap is focused on big features or focus areas; it may not call out each individual work item tha ...
 
-## Glossary
+If you have feedback or suggestions on our roadmap, please let us know.
```
This could arguably be removed, if you want to leave it in we should include *how* people should submit feedback.

EmLauber replied:
Updated to link to Github Issues' Feature Request

tnorling commented on roadmap.md:
```diff
@@ -1,60 +1,41 @@
+- Node token caching support with Redis 
+- Internal pipeline improvements for easier release cadence 
+- Deprecation of passport-azure-ad and replacement with a new lightweight token validator 
+- Updates required by the recent United State’s Executive Order on cybersecurity: Executive Order on Improving the Nation's Cybersecurity | The White House 
```
```suggestion
- Updates required by the recent United States Executive Order on cybersecurity: Executive Order on Improving the Nation's Cybersecurity | The White House 
```

tnorling replied:
Can you provide a link to the EO here as well?

EmLauber replied:
Yes that was my plan. Didn't realize the link didn't follow when I copied this out of Notebook I was working in. I will update.

tnorling commented on roadmap.md:
```diff
@@ -1,60 +1,41 @@
- 
+### Mid Term
+- Improvements on Node documentation & samples  
+- Ivy Library Distribution of MSAL-angular 
```
nit: MSAL Angular or msal-angular
```suggestion
- Ivy library distribution of MSAL Angular 
```

tnorling commented on roadmap.md:
```diff
@@ -1,60 +1,41 @@
+- Improvements on Node documentation & samples  
+- Ivy Library Distribution of MSAL-angular 
+- Angular 14 support 
+- Sustainable strategy to release adoption drivers like library dependency changes  
```
It's not clear to me what this means, can we rephrase?

tnorling commented on roadmap.md:
```diff
@@ -1,60 +1,41 @@
 
-## Past Quarters
+### Past
+This is not a comprehensive list of all completed features, but rather a way to note when something moved from near term to complete recently. You can view all past Release Notes of MSAL.js (here)[].
```
Link at the end is empty

tnorling replied:
I'd also suggest we can get by with just links to the changelogs, rather than listing out released features. Less to maintain :)

EmLauber replied:
My thinking was to move anything released from "near term" to "past" at least until the next roadmap update. My hesitancy is if a feature is removed from the list, people may not know if it was removed because we released it or because we decided not to implement it. It wouldn't be a long-lived list in past. Open to being convinced a link to release notes is enough for people to go looking for a specific feature they no longer see on the list.

tnorling replied:
I would argue that changelogs should be the one stop shop for determining what features have been released and in what version they became available. A line such as "If you don't see a feature listed on the roadmap check our changelogs to see if it has already been released or open a feature request here (link to feature request)" might help address your concerns. Thoughts?

ghost commented:
Reminder: The next release is scheduled for next week and this PR appears to be stale :( 

If changes have been requested please address feedback. 
If this PR is still a work in progress please mark as draft.

jasonnutter commented:
@peterzenz Would appreciate your review on this PR, thanks!

EmLauber committed:
define acronyms

Co-authored-by: Hector Morales <hemoral@microsoft.com>

EmLauber committed:
language updates roadmap.md

Co-authored-by: Thomas Norling <thomas.l.norling@gmail.com>

EmLauber committed:
grammar update.md

Co-authored-by: Thomas Norling <thomas.l.norling@gmail.com>

EmLauber committed:
adding executive order link

EmLauber committed:
Feedback updates

Includes adding relevant links, moving features from mid to long term, and adding how to request a new feature.

EmLauber committed:
case consistency.md

Co-authored-by: Thomas Norling <thomas.l.norling@gmail.com>

EmLauber committed:
Merge branch 'dev' into EmLauber-patch-1

tnorling commented on roadmap.md:
```diff
@@ -1,60 +1,40 @@
 
-## Past Quarters
+### Past
+This is not a comprehensive list of all completed features, but rather a way to note when something moved from near term to complete recently. You can view all past [Release Notes of MSAL.js](https://github.com/AzureAD/microsoft-authentication-library-for-js/discussions/categories/releases).
```
Links to the individual changelogs for libs may be more useful than discussions given that they are searchable and most customers may only be interested in one library or the other

EmLauber committed:
update link to release notes

EmLauber committed:
Removing past section

Directing users to changelogs instead of tracking a Past section.

EmLauber committed:
language clarity

EmLauber committed:
Merge branch 'dev' into EmLauber-patch-1

EmLauber commented:
I think I've addressed all feedback (online & offline). Looking for final reviews.

tnorling commented on roadmap.md:
```diff
@@ -1,60 +1,33 @@
+### Mid Term
+- Improvements on Node documentation & samples  
+- Ivy library distribution of MSAL Angular 
+- Angular 14 support 
```
Isn't this already done?

EmLauber replied:
I thought so but looked through release notes & didn't see it. Happy to be corrected & I missed it in release notes.

tnorling reviewed (approved):
One question, otherwise LGTM

ghost commented:
Reminder: The next release is scheduled for next week and this PR appears to be stale :( 

If changes have been requested please address feedback. 
If this PR is still a work in progress please mark as draft.

EmLauber committed:
Merge branch 'dev' into EmLauber-patch-1

EmLauber committed:
remove internal pipeline

removing internal pipeline as we already made these changes
